pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1175,6 +1175,11 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(logger klog.Logger, podIn
 
 	activated := false
 	for _, pInfo := range podInfoList {
+		// Since there may be many gated pods and they will not move from the
+		// unschedulable pool, we skip calling the expensive isPodWorthRequeueing.
+		if pInfo.Gated {
+			continue
+		}
 		schedulingHint := p.isPodWorthRequeuing(logger, pInfo, event, oldObj, newObj)
```

Comment:
How was the gated status checked before? Can we remove that check to avoid confusion with the new check?

Reply:
This was done in isPodWorthRequeueing. There is no specific logic for gated, but rather general logic for unschedulable/pending plugins

Reply:
was that by calling the PreEnqueue plugin?

@Huang-Wei, any concern with just using the `.Gated` field directly?

Reply:
Yes, the PreEnqueue plugin. That's why I added this https://github.com/kubernetes/kubernetes/pull/124669 unit test  alongside this change, to ensure that the plugin and the field stay in sync as we will now rely on derived field.

Reply:
> @Huang-Wei, any concern with just using the .Gated field directly?

No concern on my side. `.gated` is a field to store the state of calling all PreEnqueue plugins on the pod - mostly prior to adding the pod to activeQ.